### PR TITLE
[SPARK-28937][SPARK-28936][KUBERNETES] Reduce test flakyness

### PR DESCRIPTION
**Pull request by holdenk**
### What changes were proposed in this pull request?

Switch from using a Thread sleep for waiting for commands to finish to just waiting for the command to finish with a watcher & improve the error messages in the SecretsTestsSuite.

### Why are the changes needed?
Currently some of the Spark Kubernetes tests have race conditions with command execution, and the frequent use of eventually makes debugging test failures difficult.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Existing tests pass after removal of thread.sleep

### CODE REVIEW

**Comment by shaneknapp** [comment]
test this please

**Comment by shaneknapp** [comment]
(sorry for barging in again but i'm testing a small infra change on the ubuntu workers)

;)

**Comment by holdenk** [comment]
It's 100% ok :)

**Comment by shaneknapp** [comment]
> It's 100% ok :)

awesome.  i'm getting rid of those crappy, unrelated-to-the-build `lsof` errors in the logs.

**Comment by shaneknapp** [comment]
> > It's 100% ok :)
> 
> awesome. i'm getting rid of those crappy, unrelated-to-the-build `lsof` errors in the logs.

aaaaaand those error messages are gone!

`+ /home/jenkins/bin/kill_zinc_nailgun.py --zinc-port 3714`

**Comment by holdenk** [comment]
Awesome @shaneknapp :D

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SecretsTestsSuite.scala [hunk 1]
```diff
@@ -102,6 +121,8 @@ private[spark] object SecretsTestsSuite {
   val SECRET_MOUNT_PATH = "/etc/secret"
   val ENV_SECRET_KEY_1 = "username"
   val ENV_SECRET_KEY_2 = "password"
+  val ENV_SECRET_KEY_1_CAP = ENV_SECRET_KEY_1.toUpperCase(Locale.ROOT)
```
Comment (srowen) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SecretsTestsSuite.scala]: Is this part of the change related, necessary?

Comment (holdenk) [reply resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SecretsTestsSuite.scala]: No, although I figured cleaning up hard coded strings while I was there would be good.

**Review by srowen — APPROVED** [review]

File: resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/Utils.scala [hunk 2]
```diff
@@ -45,20 +48,49 @@ object Utils extends Logging {
...
+        closeLatch.await()
+      }
+    }
+    val listener = new ReadyListener()
+    val watch = pod
       .readingInput(System.in)
       .writingOutput(out)
       .writingError(System.err)
       .withTTY()
+      .usingListener(listener)
```
Comment (skonto) [inline resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/Utils.scala]: Cool I was looking for something like that in the past :)

Comment (holdenk) [reply resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/Utils.scala]: Rocking :)

**Comment by holdenk** [comment]
Merged to master.

**Pull request closed without merging**